pkg/utils/ocm.go

```diff
@@ -165,146 +150,34 @@ func GenerateQuery(clusterIdentifier string) string {
-// Taken wholesale from	openshift-online/ocm-cli
-func loadOCMConfig() (*Config, error) {
-	var err error
-
-	file, err := getOCMConfigLocation()
-	if err != nil {
-		return nil, err
-	}
-
-	_, err = os.Stat(file)
-	if os.IsNotExist(err) {
-		cfg := &Config{}
-		err = nil
-		return cfg, err
-	}
-
-	if err != nil {
-		err = fmt.Errorf("can't check if config file '%s' exists: %v", file, err)
-		return nil, err
-	}
-
-	data, err := os.ReadFile(file)
-	if err != nil {
-		err = fmt.Errorf("can't read config file '%s': %v", file, err)
-		return nil, err
-	}
-
-	if len(data) == 0 {
-		return nil, nil
-	}
-
-	cfg := &Config{}
-	err = json.Unmarshal(data, cfg)
-
-	if err != nil {
-		err = fmt.Errorf("can't parse config file '%s': %v", file, err)
-		return cfg, err
-	}
-
-	return cfg, nil
-}
-
-func getOcmConfiguration(ocmConfigLoader func() (*Config, error)) (*Config, error) {
-	tokenEnv := os.Getenv("OCM_TOKEN")
+// Creates a connection to OCM
+func CreateConnection() (*sdk.Connection, error) {
 	urlEnv := os.Getenv("OCM_URL")
-	refreshTokenEnv := os.Getenv("OCM_REFRESH_TOKEN") // Unlikely to be set, but check anyway
-
-	config := &Config{}
-
-	// If missing required data, load from the config file.
-	// We don't want to always load this, because the user might only use environment variables.
-	if tokenEnv == "" || refreshTokenEnv == "" || urlEnv == "" {
-		var fileConfigLoadError error
-		config, fileConfigLoadError = ocmConfigLoader()
-		if fileConfigLoadError != nil {
-			return config, fmt.Errorf("could not load OCM configuration file")
+	var ocmApiOverride string
+	if urlEnv != "" {
+		// if the OCM url is overridden by an env var, use that, but first we need to validate it
+		// in the case where it may be an alias
+		gatewayURL, ok := urlAliases[urlEnv]
+		if !ok {
+			return nil, fmt.Errorf("invalid OCM_URL found: %s\nValid URL aliases are: 'production', 'staging', 'integration'", urlEnv)
 		}
-	}
 
-	// Overwrite with set environment variables, to allow users to overwrite
-	// their configuration file's variables
-	if tokenEnv != "" {
-		config.AccessToken = tokenEnv
-	}
-	if urlEnv != "" {
-		config.URL = urlEnv
+		ocmApiOverride = gatewayURL
 	}
-	if refreshTokenEnv != "" {
-		config.RefreshToken = refreshTokenEnv
-	}
-
-	return config, nil
-}
-
-func CreateConnection() (*sdk.Connection, error) {
-	ocmConfigError := "Unable to load OCM config\nLogin with 'ocm login' or set OCM_TOKEN, OCM_URL and OCM_REFRESH_TOKEN environment variables"
-
-	connectionBuilder := sdk.NewConnectionBuilder()
 
-	config, err := getOcmConfiguration(loadOCMConfig)
+	config, err := ocmConfig.Load()
 	if err != nil {
-		return nil, errors.New(ocmConfigError)
+		return nil, fmt.Errorf("unable to load OCM config. %w", err)
 	}
 
-	connectionBuilder.Tokens(config.AccessToken, config.RefreshToken)
-
-	if config.URL == "" {
-		return nil, errors.New(ocmConfigError)
-	}
+	// TODO - Would be nice to put the osdctl version in here dynamically as the agent
```

Comment:
You can get that from https://github.com/openshift/osdctl/blob/8aff4f8c7efa3f8eb972199281cc3d0b55b01dcf/pkg/utils/version.go#L24 which is set via an `ldflag` at build time

Reply:
That was significantly easier than I thought it would be. Thanks!